Tests: i2s: add support for STM32H5 family

francktheb opened the pull request:
add the STM32H5 family support for the test I2s_speed

francktheb committed:
dts: arm: st: h5: add I2S nodes

Addition of I2S nodes

Signed-off-by: Franck Thebault <franck.thebault@st.com>

francktheb committed:
tests: drivers: i2s: speed: add stm32h5 support

Test on STM32H573i disco board , only the Tx transfer is ok.

Signed-off-by: Franck Thebault <franck.thebault@st.com>

erwango commented on dts/arm/st/h5/stm32h5.dtsi:
```diff
@@ -491,6 +491,45 @@
+			interrupts = <55 3>;
+			status = "disabled";
+		};
+		i2s2: i2s@40003800 {
```
Please break a line between nodes.

erwango commented on tests/drivers/i2s/i2s_speed/boards/stm32h573i_dk.overlay:
```diff
@@ -0,0 +1,61 @@
+/* i2s-node0 is the transmitter/receiver */
+
+#include <zephyr/dt-bindings/memory-attr/memory-attr-arm.h>
```
This includes doesn't look required

erwango commented on tests/drivers/i2s/i2s_speed/boards/stm32h573i_dk.overlay:
```diff
@@ -0,0 +1,61 @@
+	pinctrl-names = "default";
+	interrupts = <55 3>;
+	dmas = <&gpdma1 0 7 STM32_DMA_PERIPH_TX
+		&gpdma1 1 6 (STM32_DMA_PERIPH_TO_MEMORY | STM32_DMA_PERIPH_NO_INC |STM32_DMA_MEM_INC | STM32_DMA_PERIPH_16BITS |STM32_DMA_MEM_16BITS | STM32_DMA_PRIORITY_HIGH)>;
```
I'm surprised this long line isn't detected by CI scripts. Anyway, it will have to be shorten to be <100 chars.

Then:
- `STM32_DMA_PERIPH_RX` could be used instead of `STM32_DMA_PERIPH_TO_MEMORY | STM32_DMA_MEM_INC`
- `STM32_DMA_PERIPH_NO_INC` could be removed since it's basically 0
- What about adding a  new `STM32_DMA_16BITS` as a shorthand for `STM32_DMA_PERIPH_16BITS | STM32_DMA_MEM_16BITS`

erwango commented on tests/drivers/i2s/i2s_speed/boards/stm32h573i_dk.overlay:
```diff
@@ -0,0 +1,61 @@
+};
+
+&gpdma1 {
+	interrupts = <27 1 28 2 29 1 30 1 31 1 32 1 33 1 34 1>;
```
A comment would be nice to highlight that you're basically modifying prio here.
This being said, is that strictly required as most other priorities are 0 (which maybe we can discuss and review globally in another PR) ?

erwango commented on tests/drivers/i2s/i2s_speed/boards/stm32h573i_dk.conf:
```diff
@@ -0,0 +1,8 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+CONFIG_NOCACHE_MEMORY=n
+CONFIG_DCACHE=n
```
These 2 should already be `n` by default

francktheb committed:
drivers: dma: u5: fix dma reload

in dma_stm32_reload, the size is the number of bytes and not number
of elements.

Signed-off-by: Franck Thebault <franck.thebault@st.com>

francktheb committed:
drivers: i2s : enable rx path for H7 and H5

In i2s_stm32_isr, remove the errors from SR register
   i2s_stm32_configure, enable the rx path for H7 I2S compatible IP

Signed-off-by: Franck Thebault <franck.thebault@st.com>

jeppenodgaard commented on tests/drivers/i2s/i2s_speed/boards/stm32h573i_dk.overlay:
```diff
@@ -0,0 +1,74 @@
+	status = "okay";
+};
+
+&i2s1 {
```
Why are `interrupts` and `dma-names` defined again in the nodes?

francktheb replied:
- DMA characteristics ( (size of source (8 , 16 or 32 bits) size of destination (8 , 16 or 32 bits), the source and the destination shall be incremented or not after each DMA transfer...) depend on the application or feature (I2S, Spi, Uart...)
- - the interrupts is defined  again in the node because the priority level is set according to the involved interrupts : I2s interrupts, DMA RX and Tx interrupts....

jeppenodgaard commented on drivers/dma/dma_stm32u5.c:
```diff
@@ -545,10 +545,10 @@ static int dma_stm32_reload(const struct device *dev, uint32_t id,
 	if (stream->source_periph) {
 		LL_DMA_SetBlkDataLength(dma, dma_stm32_id_to_stream(id),
-				     size / stream->src_size);
+				     size);
```
Line length is below 100 now so 2 lines should no more be needed.

jeppenodgaard commented on dts/arm/st/h5/stm32h5.dtsi:
```diff
@@ -491,6 +491,48 @@
+			reg = <0x40003800 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00004000>,
+				<&rcc STM32_SRC_PLL1_Q SPI2_SEL(0)>;
+			dmas = <&gpdma1 0 9 STM32_DMA_PERIPH_TX
```
Still not fixed here. It's fixed the tests' overlay.
I am still not sure why almost the whole node is defined there again. The other overlays in the test does just seem to enable the node an not much more.

erwango reviewed (changes requested):
Minor comment.
Additionally:
- I'm still not clear on the IRQ priority update
- Please update PR title since no tests are involved anymore

jeppenodgaard commented on dts/arm/st/h5/stm32h5.dtsi:
```diff
@@ -491,6 +491,48 @@
+			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00004000>,
+				<&rcc STM32_SRC_PLL1_Q SPI2_SEL(0)>;
+			dmas = <&gpdma1 0 9 STM32_DMA_PERIPH_TX
+				&gpdma1 1 8 STM32_DMA_PERIPH_RX>;
```
Still not fixed here. It's fixed the tests' overlay.
I am still not sure why almost the whole node is defined there again. The other overlays in the test does just seem to enable the node an not much more.

jeppenodgaard commented on drivers/dma/dma_stm32u5.c:
```diff
@@ -544,11 +544,9 @@ static int dma_stm32_reload(const struct device *dev, uint32_t id,
 	if (stream->source_periph) {
-		LL_DMA_SetBlkDataLength(dma, dma_stm32_id_to_stream(id),
-				     size / stream->src_size);
+		LL_DMA_SetBlkDataLength(dma, dma_stm32_id_to_stream(id), size);
```
Maybe this commit should have its own PR and maybe a bug issue?

erwango commented:
@jeppenodgaard We'll issue a new PR around this
